File: learning-examples/javascript/basics/arrays.js
// Dizi İşlemleri ve Metodları

// Dizi oluşturma
let fruits = ['elma', 'armut', 'muz'];
let numbers = Array.from({length: 5}, (_, i) => i + 1);

// Dizi metodları
// push: sona eleman ekleme
fruits.push('portakal');
console.log(fruits);  // ['elma', 'armut', 'muz', 'portakal']

// pop: sondan eleman çıkarma
let lastFruit = fruits.pop();
console.log(lastFruit);  // 'portakal'

// map: her eleman için işlem
let doubled = numbers.map(num => num * 2);
console.log(doubled);  // [2, 4, 6, 8, 10]

// filter: filtreleme
let evenNumbers = numbers.filter(num => num % 2 === 0);
console.log(evenNumbers);  // [2, 4]

// reduce: toplama/birleştirme
let sum = numbers.reduce((total, num) => total + num, 0);
console.log(sum);  // 15

// find: eleman bulma
let found = fruits.find(fruit => fruit.startsWith('a'));
console.log(found);  // 'armut'

// some: koşul kontrolü
let hasLongName = fruits.some(fruit => fruit.length > 4);
console.log(hasLongName);  // true

// every: tüm elemanlar için koşul kontrolü
let allShortNames = fruits.every(fruit => fruit.length < 6);
console.log(allShortNames);  // false

// sort: sıralama
fruits.sort();
console.log(fruits);

// includes: eleman kontrolü
console.log(fruits.includes('elma'));  // true

// Dizi kopyalama
let fruitsCopy = [...fruits];  // Spread operator
let numbersCopy = Array.from(numbers); 

File: learning-examples/java/basics/Arrays.java
public class Arrays {
    public static void main(String[] args) {
        // Tek boyutlu dizi
        int[] numbers = {1, 2, 3, 4, 5};
        
        // Dizi oluşturma
        String[] fruits = new String[3];
        fruits[0] = "Elma";
        fruits[1] = "Armut";
        fruits[2] = "Muz";
        
        // Çok boyutlu dizi
        int[][] matrix = {
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        };
        
        // Dizi üzerinde iterasyon
        System.out.println("Sayılar:");
        for (int number : numbers) {
            System.out.println(number);
        }
        
        // Dizi metodları
        java.util.Arrays.sort(numbers);
        int index = java.util.Arrays.binarySearch(numbers, 3);
        
        // Matris işlemleri
        System.out.println("\nMatris:");
        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                System.out.print(matrix[i][j] + " ");
            }
            System.out.println();
        }
        
        // ArrayList kullanımı
        java.util.ArrayList<String> list = new java.util.ArrayList<>();
        list.add("Java");
        list.add("Python");
        list.add("JavaScript");
        
        System.out.println("\nProgramlama Dilleri:");
        for (String lang : list) {
            System.out.println(lang);
        }
    }
} 